### DISCUSSION TITLE
[suggestion] Ball and socket

### DISCUSSION THREAD

**Discussion by luckyrada** [tsunamayo/Starship-EVO#4001] [Suggestions]
### Short Description of the issue:

a new child entity movment system, a ball and socket, it would allow for rotation in pretty much all directions, but limited to only a certain direction, so it can turn like 60 degrees in any direction, but from then on it cant

this would help with a bunch of builds, such as a way to make small ajustments in the aim of lasers for a long range sniper or something

**Comment by tsunamayo** [comment]
I will add them for only for mech and potentially turrets. How would you supposed to control them otherwise? Cheers

**Comment by Regnion** [comment]
Ball and socket for turrets would be tremendous- this would remove the need to use a hinge and a rotor (or combination / alternation thereof) entirely.

**Comment by Pharap** [comment]
As much as I like the idea, I suspect they might be more effort to implement in code than the amount of practical value they would provide.

Anything a ball-and-socket does ought to be possible with a hinge attached to a rotor, because conceptually they both implement two axes of rotation.

That aside, the only difference I can think of is an aesthetic one. Unless there's some significant difference that hasn't occurred to me?
